syntax = "proto3";

message StartRequest {
	uint32 port = 1;
	uint32 channels = 2;
	uint32 rate = 3;
	uint32 timestamp = 4;
}

message StopRequest {
	uint32 port = 1;
	uint32 channels = 2;
}

message SensitivityRequest {
	enum Voltage {
		FIVE = 0;
		TEN = 1;
		VREF = 2;
	}
	Voltage voltage = 1;
}

message RequestWrapper {
	uint32 sequence = 1;
	oneof msg {
		StartRequest start = 2;
		StopRequest stop = 3;
		SensitivityRequest sens = 4;
	}
}
